cider-velox/src/ciderTransformer/CiderPlanPatterns.cpp

```diff
@@ -29,6 +29,9 @@ StatePtr CompoundStateMachine::Initial::accept(const VeloxPlanNodeAddr& nodeAddr
   if (auto filterNode = std::dynamic_pointer_cast<const FilterNode>(nodePtr)) {
     return std::make_shared<CompoundStateMachine::Filter>();
   } else if (auto projNode = std::dynamic_pointer_cast<const ProjectNode>(nodePtr)) {
+    if (projNode->sources()[0]->outputType()->equivalent(*(projNode->outputType()))) {
```

Comment:
Let's add TODO comments to mention this is workaround. And also mention it will be fixed once Arrow format is migrated.